lsp/src/handle.rs

```diff
@@ -26,16 +26,37 @@ struct TextDocumentChanges {
     content_changes: Vec<Text>,
 }
 
+#[derive(serde::Deserialize, Debug)]
+struct TextDocumentOpened {
```

Comment:
i don't know how to use the lsp-types properly yet, but we should try to get rid of these in the future (todo in a pr, don't worry)

Reply:
Do you mean implementing custom structs to parse the requests? or incomplete attributes?